staging/src/k8s.io/code-generator/cmd/go-to-protobuf/protobuf/namer.go

```diff
@@ -116,7 +116,10 @@ func assignGoTypeToProtoPackage(p *protobufPackage, t *types.Type, local, global
 		}
 		return
 	}
-	global[t.Name] = p
+	if t.Name.Package == p.PackagePath {
```

Comment:
isn't this wrong? Globals should be global types, i.e. cross package. You don't store those globals.

Reply:
This is correct because we track global, named types in `global` and follow the topological import order. So registering them from their own package is enough.